### ISSUE TITLE
curly: "multi-line" disallow braces on single line

### ISSUE THREAD

**Issue by UltCombo** [eslint/eslint#2053]
Currently, the [curly rule](http://eslint.org/docs/rules/curly.html) with the "multi-line" option lints both of these patterns as valid:

```js
if (foo) return;
if (foo) { return; }
```

I would like to enforce a more consistent style, that is, to always require braces on multi-line and disallow on single-line control statements. Basically, it would be identical to the current curly "multi-line" setting, except that the second pattern above should be a linting error.

Is this worth being an extra option to the core's `curly` rule (e.g. "always-multiline", "only-multiline")? Or is this more fitting for a plugin?

**Comment by UltCombo** [comment]
BTW, I can't use curly's "multi" setting because I want multi-line single statement to pass:

```js
if (foo) {
    return;
}
```

So the presence of braces should be mandated by the number of lines, not the number of statements.

**Comment by nzakas** [comment]
I think we can work it in.

**Comment by rightaway** [comment]
+1

**Comment by alberto** [comment]
Closing due to lack of activity.

**Comment by UltCombo** [comment]
I don't have time to implement this feature at the current time.
This feature-request seems to have good acceptance, but well, do whatever you please.
